model-optimizer/mo/ops/pooling.py

```diff
@@ -53,7 +72,15 @@ def backend_attrs(self):
 
     @staticmethod
     def infer(node: Node):
-        assert (len(node.in_nodes()) == 1)
+        Pooling.pool_infer(node)
+
+    @staticmethod
+    def pool_infer(node: Node, is_pool_v2=False):
+        if is_pool_v2:
+            assert (len(node.in_nodes()) == 3)
```

Comment:
I recommend to move these checks under `infer` method of operation itself. And move `window` and `stride` values to attributes as well. It is because solution is not extendible for v3-... cases.  And make `pool_infer` to be the common function and not a static method of `Pool`

Reply:
It is our standard approach to keep infer method as part of the Op class. Because it should be used in the Op only. So I don't see any reason to move it to common functions. But I agree that the introducing parameter `is_pool_v2` is not a scalable solution.

Reply:
Moved FW version specific logic to the corresponding infer functions. Now common base infer function `pool_infer` is version agnostic. Please review